theano/gpuarray/opt.py

```diff
@@ -235,10 +250,10 @@ def local_opt(node):
                     elif isinstance(new_op, (tuple, list)):
                         return [safe_to_cpu(o) for o in new_op]
                     else:  # suppose it is a variable on the GPU
-                        return [host_from_gpu(new_op)]
+                        return [new_op.transfer('cpu')]
             return False
         local_opt.__name__ = maker.__name__
-        return local_optimizer(OP)(local_opt)
+        return register_topo('fast_compile')(local_optimizer(OP)(local_opt))
```

Comment:
Currently, as we don't have a backward pass, if we need it, we will see a big time inside the gpu_optimizer Equilibrium that will do an inefficient backward pass. The backward pass is just to speed up the compilation as it is currently done inefficiently. I didn't see a profile that show this is taking significant time.